### PR TITLE
Refactor dublin core utility

### PR DESCRIPTION
**Pull request by johannes-manner**
The tool is now implemented as a console application with a read-evaluate-print cycle.
It can be used with and without a jar file and can be used independently of #3704.

----

- [x] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [x] Manually tested changed features in running JabRef
- [x] Check documentation status (Updated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/org/jabref/cli/XmpUtilMain.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        System.out.println("-------------------------------------------------");
+        System.out.print("Choose an option: ");
+    }
+
+    /**
+     * The tool is implemented as a console application with a read-evaluate-print cycle.
+     *
+     * @param args
+     */
+    public static void main(String[] args) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/cli/XmpUtilMain.java]: Like the idea, but why not integrate it with the rest of our cli infrastruture? 
Maybe as part of an "Importer" or Exporter?
http://help.jabref.org/en/CommandLine#import-file--i-filenameimport-format

File: src/main/java/org/jabref/cli/XmpUtilMain.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        System.out.println("(2) Write entries in bib file to Pdf metadata");
+        System.out.println("    To report bugs visit https://issues.jabref.org");
+        System.out.println("-------------------------------------------------");
+        System.out.print("Choose an option: ");
+    }
+
+    /**
+     * The tool is implemented as a console application with a read-evaluate-print cycle.
+     *
+     * @param args
```
Comment (koppor) [inline src/main/java/org/jabref/cli/XmpUtilMain.java]: Remove empty parameter description 😇

**Comment by koppor** [comment]
Regarding integration in JabRef: In https://github.com/JabRef/jabref/pull/3704, I separated the main method completely to be able to offer a non-fat jar...

However, I would agree to integrate everything in our normal CLI and then, in the context of the Java 9 refactorings (https://github.com/JabRef/jabref/pull/3421) create finer-grained compilation units.

**Comment by johannes-manner** [comment]
I integrated the xmp functionality in the CLI infrastructure :)

After exiting my tool with '0' the graphical user interface starts... 
@Siedlerchr  How can I suppress the GUI?

**Comment by Siedlerchr** [comment]
When I am right, you just need to call Platform.exit() (at least when it's the javafx application thread start )
Otherwise System.exit(0) (shuts down the Java vm and signals a success)

**Comment by johannes-manner** [comment]
Platform.exit() does not work. Therefore I used the other option System.exit(0).

The implementation is now ready for review. 
Where can I write the help entry for the http://help.jabref.org/en/CommandLine webpage?

**Comment by Siedlerchr** [comment]
@johannes-manner  Okay, good to hear! Help files are all in their own Repo. Either you click on that Octocat Github thing in the right corner, or you go to that repo:
https://github.com/JabRef/help.jabref.org/blob/gh-pages/en/CommandLine.md

File: src/main/java/org/jabref/cli/XmpUtilMain.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                    String pdfFile = consoleReader.readLine().trim();
+                    XmpUtilMain.writeBibFileToPdfMetadata(bibFile, pdfFile);
+                }
+            } catch (IOException | TransformerException e) {
+                System.err.println(e.getMessage());
+            }
+        }
+    }
+
+    public static void main(String[] args) {
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/cli/XmpUtilMain.java]: Do we actually need this tool as a separate executable now that the functionality is included in the usual CLI?

Comment (johannes-manner) [reply src/main/java/org/jabref/cli/XmpUtilMain.java]: It's a convenient tool to add metadata to pdf files. If there is the possiblity to build a standalone jar with the multi-model build, why not use the chance to offer this tool to more users and make this tool more popular?

**Pull request merged**